### PR TITLE
Eliminated warnings and concerning "INFO"s when loading schema

### PR DESCRIPTION
**Pull request by turbomam**
#428

### CODE REVIEW

**Comment by turbomam** [comment]
- moved some comments outside of the `target/python/%.py` Makefile rule 
- sorted `imports`, `prefixes` and `default_curi_maps`
  - **ordering of `default_curi_maps` might be significant**
- some prefixes are still set to expand to http://example.com/
- all schema modules now have `default_range` and `default_prefix`
  - **nmdc prefix isn't resolvable**
  - **mixs/MIXS prefix needs work too.**
  - **prov?**. esp default_prefix
- moved some usage-only slots out to the module's outer slots: block
  - like `has part` in `src/schema/annotation.yaml`  , several in `src/schema/nmdc.yaml`, and more
  - note that this was also a rename operation from `has_part` to `has part`, to be consistent with `part of`, although I would like to normalize all slots to lower snake case

**Comment by turbomam** [comment]
- ./annotation.yaml:  COG: http://example.com/
- ./annotation.yaml:  EC: http://example.com/
- ./annotation.yaml:  MetaCyc: http://example.com/
- ./annotation.yaml:  MetaNetX: http://example.com/
- ./annotation.yaml:  RetroRules: http://example.com/
- ./core.yaml:  UniProtKB: http://example.com/
- ./core.yaml:  gtpo: http://example.com/
- ./nmdc.yaml:  COG: http://example.com/
- ./nmdc.yaml:  EC: http://example.com/
- ./nmdc.yaml:  ISA: http://example.com/
- ./nmdc.yaml:  MetaCyc: http://example.com/
- ./nmdc.yaml:  MetaNetX: http://example.com/
- ./nmdc.yaml:  RetroRules: http://example.com/
- ./nmdc.yaml:  UniProtKB: http://example.com/
- ./nmdc.yaml:  gtpo: http://example.com/
- ./nmdc.yaml:  insdc.srs: http://example.com/
- ./nmdc.yaml:  mgnify: http://example.com/

grep -R ': https://example.com' | grep -v 'from_schema:' ? none

**Comment by turbomam** [comment]
https://github.com/perma-id/w3id.org/blob/master/nmdc/.htaccess

> Options +FollowSymLinks
RewriteEngine on
RewriteRule ^(.*) https://github.com/microbiomedata/ [R=307,L]

So https://w3id.org/nmdc/Study -> https://github.com/microbiomedata/

But should -> https://microbiomedata.github.io/nmdc-schema/Study/

Anyway, the nmdc prefix should expand to https://w3id.org/nmdc/, at least on a term-by-term basis.

It is currently is set to 'https://microbiomedata/meta/'

**Comment by turbomam** [comment]
Name-based MIxS docs page: https://genomicsstandardsconsortium.github.io/mixs/Water/

There has been some talk of switching to id-based pages. Could we have a hybrid within the NMDC schema and it's docs pages? Name base pages for NMDC and id-based for MIxS?

Is this redirection coming from either or both?
- https://github.com/perma-id/w3id.org/blob/master/mixs/.htaccess
- https://github.com/perma-id/w3id.org/blob/master/gensc/.htaccess

try 
- https://w3id.org/mixs/Water -> https://genomicsstandardsconsortium.github.io/mixs/Water/
- ~~https://w3id.org/gensc/Water~~
- ~~https://w3id.org/gensc/mixs/Water~~

therefore, prefix `mixs` should expand to https://w3id.org/mixs

Also, since the w3id namespace is lowercase `mixs`, shouldn't the corresponding LinkML prefix be lowercase, too? Check the NMDC and MIxS schemas for usage of the uppercase variant `MIXS`

**Comment by turbomam** [comment]
This claimed mapping in `src/schema/mixs.yaml` doesn't currently exist:

```yaml
  env_package:
    name: env_package
    aliases:
    - environmental package
    mappings:
    - mixs:env_package
```

File: src/schema/annotation.yaml [hunk 1]
```diff
@@ -4,25 +4,36 @@ title: Annotation Module for NMDC Schema
 description: >-
   This module in the schema is for representing annotations including functional annotations of proteins and other gene products,
   as well as controlled terms for describing things like metabolites
+default_range: string
 
 see_also:
   - https://github.com/microbiomedata/nmdc-metadata/issues/176
 
 default_curi_maps:
-  - obo_context
```
Comment (cmungall) [inline src/schema/annotation.yaml:12]: what's the reason for this? obo should take precedence

File: src/schema/annotation.yaml [hunk 2]
```diff
@@ -4,25 +4,36 @@ title: Annotation Module for NMDC Schema
...
   - idot_context
+  - obo_context
 
 prefixes:
-  nmdc: https://microbiomedata/meta/
+  COG: http://example.com/
```
Comment (cmungall) [inline src/schema/annotation.yaml:17]: if we're going to make this change we should use the correct URLs

I think the linter will suggest the right URLs, cc @pkalita-lbl

Comment (pkalita-lbl) [reply src/schema/annotation.yaml]: The `canonical_prefixes` rule is a pretty light-weight wrapper around https://github.com/linkml/prefixmaps. You can provide the `prefixmaps_contexts` option as a list of strings that will be passed to prefixmaps' `load_multi_context` to prioritize some prefix registries over others (the prefixmaps docs can explain it better than me 😁). But that's all there is in terms of configuration for now. What else would be helpful for you?

**Review by sujaypatil96 — APPROVED** [review]

**Pull request merged**